merlin/loader/ops/embeddings/torch_embedding_op.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+import torch
+from torch.nn import Embedding
+
+from merlin.core.protocols import Transformable
+from merlin.dag import BaseOperator
+from merlin.dag.selector import ColumnSelector
+from merlin.schema import ColumnSchema, Schema, Tags
+
+
+class TorchEmbeddingOperator(BaseOperator):
+    """Create an operator that will apply a torch embedding table to supplied indices.
+    This operator allows the user to supply an id lookup table if the indices supplied
+    via the id_lookup_table.
+
+    Parameters
+    ----------
+    embeddings : np.ndarray
+        numpy ndarray representing embedding values
+    lookup_key : str, optional
+        the name of the column that will be used as indices, by default "id"
+    embedding_name : str, optional
+        name of new column of embeddings, added to output, by default "embeddings"
+    id_lookup_table : np.array, optional
+        numpy array of values that represent embedding indices, by default None
+    """
+
+    def __init__(
+        self,
+        embeddings: np.ndarray,
+        lookup_key: str = "id",
+        embedding_name: str = "embeddings",
+        id_lookup_table=None,
+    ):
+        self.embeddings = (
+            embeddings
+            if isinstance(embeddings, Embedding)
+            else Embedding.from_pretrained(torch.FloatTensor(embeddings))
+        )
+        self.lookup_key = lookup_key
+        self.embedding_name = embedding_name
+        self.id_lookup_table = id_lookup_table
+
+    def transform(
+        self, col_selector: ColumnSelector, transformable: Transformable
+    ) -> Transformable:
+        keys = transformable[self.lookup_key]
+        indices = keys.cpu()
+        if self.id_lookup_table is not None:
+            indices = torch.Tensor(np.nonzero(np.in1d(self.id_lookup_table, indices))).to(
+                torch.int32
+            )
+        embeddings = self.embeddings(indices)
+        transformable[self.embedding_name] = torch.squeeze(embeddings.to(keys.device))
+        return transformable
+
+    def compute_output_schema(
+        self,
+        input_schema: Schema,
+        col_selector: ColumnSelector,
+        prev_output_schema: Schema = None,
+    ) -> Schema:
+        """Creates the output schema for this operator.
+
+        Parameters
+        ----------
+        input_schema : Schema
+            schema coming from ancestor nodes
+        col_selector : ColumnSelector
+            subselection of columns to apply to this operator
+        prev_output_schema : Schema, optional
+            the output schema of the previously executed operators, by default None
+
+        Returns
+        -------
+        Schema
+            Schema representing the correct output for this operator.
+        """
+        col_schemas = []
```

Comment:
It seems like this code is almost identical to the TF code except for how the dtype is fetched. Could it be extracted out into a helper method that accepts the dtype as a parameter?